synth_node_bin/src/main.rs

```diff
@@ -0,0 +1,128 @@
+//! A synthetic node binary can be used to interact with the node in the
+//! background from a different runtime environment.
+//!
+//! This is a quick solution - in future Ziggurat projects it will be done differently.
+//!
+//! Remaining worklist (probably not going to be implemented in this repo):
+//!   - Add an argument option to choose specific action in order to support
+//!     different synthetic node binary implementations at once. A few examples:
+//!     ```
+//!        ./synthetic_node_bin --action=A    // Runs an idle/friendly synthetic node
+//!        ./synthetic_node_bin --action=B    // Runs a wild synthetic node which does something funny
+//!     ```
+use std::{net::SocketAddr, process::ExitCode};
+use tokio::time::{sleep, Duration};
+
+use clap::Parser;
+use ziggurat_xrpl::tools::{config::SynthNodeCfg, synth_node::SyntheticNode};
+use anyhow::Result;
+
+/// A synthetic node which can connect to the XRPL node and preform some
+/// actions independently.
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct CmdArgs {
+    /// An address of the node in the <ip>:<port> format.
+    #[arg(short = 'i', long)]
+    node_ip: Option<SocketAddr>,
+
+    /// Always reconnect in the case the connection fails - synthetic node never dies.
+    #[arg(short = 's', long, default_value_t = false)]
+    stubborn: bool,
+
+    /// Enable tracing.
+    #[arg(short = 't', long, default_value_t = false)]
+    tracing: bool,
+}
+
+#[tokio::main]
+async fn main() -> ExitCode {
+    let args = CmdArgs::parse();
+
+    let node_addr = if let Some(addr) = args.node_ip {
```

Comment:
Maybe it would be shorter:
```
    let node_addr = args.node_ip.unwrap_or_else(|| {
        eprintln!("Node address should be provided.");
        return ExitCode::FAILURE;
    });
```

Reply:
I like the idea, but we can't use the `return` statement in closure to return the code from the `main` fn.

I'm not sure if there is some sort of alternative solution out there yet for this particular situation - to make it simpler (like your code snippet)